lib/Service/Mail/Mailer.hs

```diff
@@ -6,58 +6,89 @@ module Service.Mail.Mailer
       body =
-        TL.pack $ "Hi! We have a new user (" ++ uName ++ " " ++ uSurname ++ ", " ++ uEmail ++
-        ") in our Wizard! Elixir DSW Wizard Team"
-  createEmail analyticsAddress subject body
+        TL.pack $ "Hi! We have a new user (" ++ uName ++ " " ++ uSurname ++ ", " ++ uEmail ++ ") in our Wizard! " ++
+        mailName ++
+        " Team"
+  sendEmail analyticsAddress subject body
 
 sendResetPasswordMail :: Email -> U.UUID -> String -> AppContextM ()
 sendResetPasswordMail email userId hash = do
   dswConfig <- asks _appContextConfig
   let clientAddress = dswConfig ^. clientConfig . address
       clientLink = clientAddress ++ "/forgotten-password/" ++ U.toString userId ++ "/" ++ hash
       link = "<a href=\"" ++ clientLink ++ "\">here</a>"
-      subject = "Reset Password"
-      body = TL.pack $ "Hi! You can set up a new password " ++ link ++ "! Elixir DSW Wizard Team"
-  createEmail email subject body
+      mailName = dswConfig ^. mail . name
+      subject = TL.pack $ mailName ++ ": Reset Password"
+      body = TL.pack $ "Hi! You can set up a new password " ++ link ++ "! " ++ mailName ++ " Team"
+  sendEmail email subject body
+
+-- --------------------------------
+-- PRIVATE
+-- --------------------------------
+createEmail to from subject plainBody "" _ = MIME.simpleMail' to from subject plainBody
+createEmail to from subject plainBody htmlBody attachments =
+  MIME.simpleMailInMemory to from subject plainBody htmlBody attachments
+
+makeConnection False host Nothing = SMTP.doSMTP host
+makeConnection False host (Just port) = SMTP.doSMTPPort host (fromIntegral port)
+makeConnection True host Nothing = SMTPSSL.doSMTPSSL host
+makeConnection True host (Just port) = SMTPSSL.doSMTPSSLWithSettings host settings
+  where
+    settings = SMTPSSL.defaultSettingsSMTPSSL {SMTPSSL.sslPort = fromIntegral port}
+
+sendEmail :: Email -> TL.Text -> TL.Text -> AppContextM ()
+sendEmail to subject body = do
+  dswConfig <- asks _appContextConfig
+  let mailConfig = dswConfig ^. mail
+      from = mailConfig ^. email
+      addrFrom = MIME.Address (Just . T.pack $ mailConfig ^. name) (T.pack from)
+      addrTo = MIME.Address Nothing (T.pack to)
+      plainBody = ""
+      htmlBody = body
+      mailHost = mailConfig ^. host
+      mailPort = mailConfig ^. port
+      mailSSL = mailConfig ^. ssl
+      mailUsername = mailConfig ^. username
+      mailPassword = mailConfig ^. password
+      mailSubject = TL.toStrict subject
+      mailMessage = createEmail addrFrom addrTo mailSubject plainBody htmlBody []
+  if mailConfig ^. enabled
+    then liftIO $ makeConnection mailSSL mailHost mailPort $ \connection -> do
+           authSuccess <- SMTP.authenticate Auth.LOGIN mailUsername mailPassword connection
+           renderedMail <- MIME.renderMail' mailMessage
+           if authSuccess
+             then SMTP.sendMail from [to] (S.concat . B.toChunks $ renderedMail) connection
+             else return ()
```

Comment:
We should log when email was sent (probably just as an INFO) and when it failed (I would log it as a WARNING)

Reply:
Please, just try to get rid off runStdoutLoggingT here

Reply:
Done in https://github.com/ds-wizard/dsw-server/pull/42/commits/9c0744920269af18c9534711948820699f4c8541 (other errors, such as errors when connecting to the server, are logged as well with error message from IO exception). 

Rollbacks in [DSW-205](https://ds-wizard.atlassian.net/browse/DSW-205)